public/app/features/explore/Logs.tsx

```diff
@@ -1,29 +1,130 @@
 import React, { Fragment, PureComponent } from 'react';
 import Highlighter from 'react-highlight-words';
 
+import { RawTimeRange } from 'app/types/series';
 import { LogsModel } from 'app/core/logs_model';
 import { findHighlightChunksInText } from 'app/core/utils/text';
+import { Switch } from 'app/core/components/Switch/Switch';
+
+import Graph from './Graph';
+
+const graphOptions = {
+  series: {
+    bars: {
+      show: true,
+    },
+  },
+  yaxis: {
+    tickDecimals: 0,
+  },
+};
 
 interface LogsProps {
   className?: string;
   data: LogsModel;
   loading: boolean;
+  position: string;
+  range?: RawTimeRange;
```

Comment:
Is there a reason your using RawTimeRange instead of TimeRange (which includes the raw as well). Raw time range requires parsing to be useful as to and from can be strings like "now", "now-1h" etc.

Reply:
I'm using `RawTimeRange` here because it's being used all over Explore. Mostly because of the string nature of its members, which works well for controlled components. I tend to parse the raw strings only when needed and then produce one-off objects that I have to maintain. Keeping `TimeRange` properties in sync is a bit of a hassle at the moment.

Reply:
what do you mean keeping TimeRange properties in sync? If you want the string based time values you can they are in the raw property on the TimeRange interface. If you use that each component does not need to parse the time range (Graph for example) if it needs the moment value. 

The common Graph component (that panel & explore will use once it's more complete) is using TimeRange:
https://github.com/grafana/grafana/blob/master/public/app/viz/Graph.tsx#L61

Reply:
I agree that it's suboptimal. Explore is consuming what the Explore-specific time picker produces, and that's `RawTimeRange`. I propose to fix this in a different PR.